src/Sankey.js

```diff
@@ -0,0 +1,201 @@
+import {sankey, sankeyLinkHorizontal} from "d3-sankey";
+
+import {accessor, assign, configPrep, constant, elem} from "d3plus-common";
+import {Path, Rect} from "d3plus-shape";
+import {dataLoad as load, Viz} from "d3plus-viz";
+
+/**
+    @class Rings
+    @extends external:Viz
+    @desc Creates a sankey visualization based on a defined set of XXXXX. [Click here](xxxxxx) for help getting started using the Sankey class.
+*/
+export default class Sankey extends Viz {
+
+  /**
+      @memberof Sankey
+      @desc Invoked when creating a new class instance, and sets any default parameters.
+      @private
+  */
+  constructor() {
+    super();
+    this._nodeId = false;
+    this._links = accessor("links");
+    this._noDataMessage = false;
+    this._nodes = accessor("nodes");
+    this._nodeWidth = 30;
+    this._sankey = sankey();
+    this._shapeConfig = assign(this._shapeConfig, {
+      labelConfig: {
+        duration: 0,
+        fontMin: 1,
+        fontResize: true,
+        labelPadding: 0,
+        textAnchor: "middle",
+        verticalAlign: "middle"
+      },
+      Path: {
+        fill: "none",
+        label: false,
+        stroke: "#dbdbdb",
+        strokeWidth: 20,
+        strokeOpacity: 0.4
+      }
+    });
+  }
+
+  /**
+      Extends the draw behavior of the abstract Viz class.
+      @private
+  */
+  _draw(callback) {
+    super._draw(callback);
+
+    const height = this._height - this._margin.top - this._margin.bottom,
+          width = this._width - this._margin.left - this._margin.right;
+
+    const path = sankeyLinkHorizontal();
+
+    const nodes = this._nodes;
+
+    const nodeLookup = this._nodeLookup = this._nodes.reduce((obj, d, key) => {
+      obj[d[this._nodeId || "id"]] = key;
+      return obj;
+    }, {});
+
+    const links = this._links.map(link => {
+      const check = ["source", "target"];
+
+      const linkLookup = check.reduce((result, item) => {
+        result[item] =
+          typeof link[item] === "number" ? link[item] : nodeLookup[link[item]];
+        return result;
+      }, {});
+
+      return {
+        source: linkLookup.source,
+        target: linkLookup.target,
+        value: link.value
+      };
+    });
+
+    const transform = `translate(${this._margin.left}, ${this._margin.top})`;
+
+    this._sankey
+      .nodeWidth(this._nodeWidth)
+      .nodes(nodes)
```

Comment:
does running this add values to the user defined `nodes` list? If so, we should be nesting the user data inside of a custom object so that we don't modify their code (we always want to be non-desctructive to user-passed data):

example internal node:
```js
{
  __d3plus__: true,
  data: d,
  i,
  id: this._nodeId(d, i),
  x: d.x.
  y: d.y
}